### ISSUE TITLE
Develop "Data Extract" reports for Elements

### ISSUE THREAD

**Issue by DevinSmithWork** [eScholarship/lbl-recent-pubs-report#3]
### Why 

Campus partners would like to benefit from the Reporting Database in a convenient manner, such as by generating detailed reports and data extracts from within the UCPMS. 

### What 

Please develop data extract reports for campus partner use in the UCPMS. An initial list of suggested reports is below. 

TIP: Every data extract report has an option to download its .SQL definition file. 

* existing custom data extract: [LBNL labels associated with claimed publications](https://oapolicy.universityofcalifornia.edu/reportdetails.html?id=179) (shamelessly based on [SDG report](https://oapolicy.universityofcalifornia.edu/reportdetails.html?id=129))

* Sympletic Support: [How to make a custom Data Extract available](https://support.symplectic.co.uk/support/solutions/articles/6000257102)

#### 1. Group OA Policy Stats Summary 

[Convert the systemwide OAPI "campus groups: text" report](https://public.tableau.com/views/UCCampusOAPolicyImplementationDashboard/CampText) to a Data Extract 

Expectation: A Campus Admin should be able to generate the report at the Campus, Lbl 3, Lvl 4, Lvl 5, etc. If they've only chosen one department (Lvl 1), then there would be one row of data; the first few columns should define the various levels (UC Berkeley > UC Berkeley All Faculty > Berkeley Law > Berkeley Law). Campuses should be able to query any groups under the main campus group (i.e. also the primary groups, reporting groups). 

#### 2. Pending publications

Campuses have expressed that they would like a list of "Pending publications" for individual users or specified groups, and like the formatting of the "Publications summary" report,  https://oapolicy.universityofcalifornia.edu/reportdetails.html?id=128

To reduce the bloat in what will undoubtedly be a massive report,  let's reduce the columns to:
 
* UCPMS ID [formatted as profile URL]
* User's full name in alphabetical format
* User's email address 
* Primary-group 
* Is current user (T/F)
* Publication type 
* Pub date 
* Record date 1
* DOI 
* Title 
* Journal/Conference 
* status in eScholarship [may have multiple]
* eScholarship URL [may have multiple]

#### 3. OA Policy publications 

Campuses would like a version of the "Publications summary" report that applies only to claimed in OA Policy articles,  https://oapolicy.universityofcalifornia.edu/reportdetails.html?id=128

* UCPMS ID [formatted as profile URL]
* User's full name in alphabetical format
* User's email address 
* Primary-group 
* Is current user (T/F)
* Publication type 
* Pub date 
* Record date 1
* DOI 
* Title 
* Journal / Conference 
* publication OA status 
* RGPO grant links (grant IDs; concat)
* LBL grant links (grant IDs; concat)
* UCD grant links (grant IDs; concat)
* other grant links (grant IDs; concat)
* status in eScholarship (may have multiple)
* eScholarship URL (may have multiple) 
* embargo date (if any)
* OA Policy exception (if any)

#### 4. Users: Publication counts, delegates, search settings 

Campuses can use these reports to monitor many things at the group level, such as identifying outliers and monitoring departmental OA Policy participation. 

Convert ["users" report](https://drive.google.com/file/d/1f0Z5Ock-qte7DHda_V2qReK-koRDeq_b/view?usp=drive_link) to Elements data extract 

updates needed: 

* add column "OA Policy", data from OA Policy Name
* make UCPMS ID a link to profile URL

**Comment by DevinSmithWork** [comment]
No. 2, 3, and 4:

Pending Publications by Group or Users
https://oapolicy.universityofcalifornia.edu/reportdetails.html?id=181

OA Policy Publications by Group or Users
https://oapolicy.universityofcalifornia.edu/reportdetails.html?id=182

User Reports (adapted from the UCD query)
https://oapolicy.universityofcalifornia.edu/reportdetails.html?id=184

Working on No. 1, it'll require a bit more modification.

**Comment by DevinSmithWork** [comment]
Rewrite of the UCD query, using joins rather than subqueries. (Also added the requested profile links & OA Policy names): https://oapolicy.universityofcalifornia.edu/reportdetails.html?id=193

**Comment by DevinSmithWork** [comment]
No. 1 -- The Campus Public Tableau text version report: https://oapolicy.universityofcalifornia.edu/reportdetails.html?id=194

Note: I've limited the groups-for-querying to the set appropriate for the query. It's using the single-group parameter, so only one group is able to be selected at a time.

**Comment by alainna** [comment]
> Pending Publications by Group or Users
> https://oapolicy.universityofcalifornia.edu/reportdetails.html?id=181

Runs surprisingly well on a single department (even if that dept is STEM). 

Let's remove the "Verification Status" column, and add a "Deposited" or "Repository Status" column which indicates the current status in eScholarship. 

---- 

> OA Policy Publications by Group or Users
> https://oapolicy.universityofcalifornia.edu/reportdetails.html?id=182

Let's remove: 

* "Verification Status" columns
* "User Proprietary ID" column
* "Primary group descriptor" column (retain "Primary group")
* "Publication Status"

and add: 

* "Deposited" or "Repository Status" column which indicates the current status in eScholarship 

---- 

> User Reports (adapted from the UCD query)
https://oapolicy.universityofcalifornia.edu/reportdetails.html?id=184

Worked! On both smaller (UCD Vet school department) and somewhat larger (UCB senate primary group) groups. Haven't tried a group larger than 8000 yet. 


> Rewrite of the UCD query, using joins rather than subqueries. (Also added the requested profile links & OA Policy names): https://oapolicy.universityofcalifornia.edu/reportdetails.html?id=193

I haven't been able to successfully run this one. 🙁 Any tips or ones that've worked?? 

example error-


``` 
Error: Rendering report 'User Details, Search Settings, and Pub Counts by Groups or Users' for System Administrator

[...]

Level 1:

-------

Exception type: System.Data.SqlClient.SqlException

Message: Could not allocate space for object '<temporary system object: 453539852517376>' in database 
'tempdb' because the 'PRIMARY' filegroup is full due to lack of storage space or database files reaching the 
maximum allowed size. Note that UNLIMITED files are still limited to 16TB. Create the necessary space by 
dropping objects in the filegroup, adding additional files to the filegroup, or setting autogrowth on for 
existing files in the filegroup.
[...]

SQL-specific information:

SQL Error:
	Class: 17
	Number: 1105
	State: 2
	Provider: .Net SqlClient Data Provider
	Server: localhost
	Procedure: 
	Line number: 12
```

----- 

> No. 1 -- The Campus Public Tableau text version report: https://oapolicy.universityofcalifornia.edu/reportdetails.html?id=194

Seems to have been running in the background a while for UCB, but no error yet. Will update when it generates.

**Comment by DevinSmithWork** [comment]
Requested modifications made for

> Pending Publications by Group or Users
> https://oapolicy.universityofcalifornia.edu/reportdetails.html?id=181

> OA Policy Publications by Group or Users
> https://oapolicy.universityofcalifornia.edu/reportdetails.html?id=182

The big UCD users, pub counts, and search settings query has been rewritten for efficiency. Takes ~2 mins for a group with 40 members. The runtime may expand more-than-linearly, though: https://oapolicy.universityofcalifornia.edu/reportdetails.html?id=193

The error you included for this query is SQL engine-level stuff that we almost certainly don't have permissions to alter on the reporting DB. However, if we run into this enough, it may be worthwhile flagging it for Symplectic to get them to make adjustments.

**Comment by alainna** [comment]
Confirmed, with thanks!! I'm going to flag these as completed :)
